infrastructure/src/main/kotlin/extension/RedmadrobotExtension.kt

```diff
@@ -119,3 +127,17 @@ public class TestOptions {
         useJunitPlatform = false
     }
 }
+
+public class DetektOptions {
+
+    /** Option for turn on check only changed files */
+    internal var checkDiffOnly: Boolean = false
+
+    /** Target branch to compare changes */
+    internal var targetDiffBranch: String = ""
+
+    public fun checkOnlyDiffOnBranch(branch: String) {
```

Comment:
Public functions should be documented

Reply:
Also, it would be great to add branch name validation here because this string provided by a user.
```kotlin
require(branch.isNotBlank()) { "..." }
```

Reply:
Added validation.